### PR TITLE
Pin pyyaml to latest 4.x

### LINKED ISSUE #106: Build failing due to yaml parsing in 9.0/10.0/11.0
On master, the build is failing with:

```
2019-03-11 08:21:55,103 79 CRITICAL odoodb_test odoo.service.server: Failed to initialize database `odoodb_test`.
Traceback (most recent call last):
  File "/odoo/src/odoo/service/server.py", line 945, in preload_registries
    registry = Registry.new(dbname, update_module=update_module)
  File "/odoo/src/odoo/modules/registry.py", line 85, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/odoo/src/odoo/modules/loading.py", line 321, in load_modules
    report=report, models_to_check=models_to_check)
  File "/odoo/src/odoo/modules/loading.py", line 184, in load_module_graph
    _load_data(cr, module_name, idref, mode, kind='demo')
  File "/odoo/src/odoo/modules/loading.py", line 95, in _load_data
    tools.convert_file(cr, module_name, filename, idref, mode, noupdate, kind, report)
  File "/odoo/src/odoo/tools/convert.py", line 789, in convert_file
    convert_yaml_import(cr, module, fp, kind, idref, mode, noupdate, report)
  File "/odoo/src/odoo/tools/yaml_import.py", line 854, in yaml_import
    yaml_interpreter.process(yaml_string)
  File "/odoo/src/odoo/tools/yaml_import.py", line 794, in process
    for node in yaml.load(yaml_string):
  File "/usr/local/lib/python3.5/dist-packages/yaml/__init__.py", line 113, in load
    return loader.get_single_data()
  File "/usr/local/lib/python3.5/dist-packages/yaml/constructor.py", line 43, in get_single_data
    return self.construct_document(node)
  File "/usr/local/lib/python3.5/dist-packages/yaml/constructor.py", line 52, in construct_document
    for dummy in generator:
  File "/usr/local/lib/python3.5/dist-packages/yaml/constructor.py", line 404, in construct_yaml_map
    value = self.construct_mapping(node)
  File "/usr/local/lib/python3.5/dist-packages/yaml/constructor.py", line 210, in construct_mapping
    return super().construct_mapping(node, deep=deep)
  File "/usr/local/lib/python3.5/dist-packages/yaml/constructor.py", line 131, in construct_mapping
    key = self.construct_object(key_node, deep=deep)
  File "/usr/local/lib/python3.5/dist-packages/yaml/constructor.py", line 92, in construct_object
    data = constructor(self, node)
  File "/usr/local/lib/python3.5/dist-packages/yaml/constructor.py", line 420, in construct_undefined
    node.start_mark)
yaml.constructor.ConstructorError: could not determine a constructor for the tag '!record'
  in "<byte string>", line 2, column 3:
      !record {model: 'res.partner', i ... 
      ^
```

### PR DESCRIPTION
**Pull request by tschanzt**
Fixes #106

### CODE REVIEW

File: 12.0/base_requirements.txt [hunk 1]
```diff
@@ -81,6 +81,6 @@ ptyprocess==0.5.2
 py==1.5.3
 pycodestyle==2.3.1
 pyflakes==1.6.0
-PyYAML>=4.2b1
+PyYAML==4.2b4
```
Comment (sebalix) [inline 12.0/base_requirements.txt]: It is worth to test the 5.1 release on v12, as Odoo doesn't use PyYaml anymore we have no more conflict on this.

Comment (tschanzt) [reply 12.0/base_requirements.txt]: i guess we could but i think there is also value in having the same version in all templates so you don't have to deal with different bugs. @guewen what do you think.

Comment (guewen) [reply 12.0/base_requirements.txt]: I'd prefer to keep older versions on the older odoo versions but upgrade to newer versions as we move to more recent odoo versions. Make sense to me to try 5.1 on 12.0

Comment (p-tombez) [reply 12.0/base_requirements.txt]: Is it still relevant to have this lib in v12 as Odoo discontinued the usage of Yaml files ?

Comment (sebalix) [reply 12.0/base_requirements.txt]: @p-tombez it is a dependency of marabunta so yes

Comment (guewen) [reply 12.0/base_requirements.txt]: > Is it still relevant to have this lib in v12 as Odoo discontinued the usage of Yaml files ?

No. Probably not much used in OCA addons neither. 

In marabunta, we move to ruamel.yaml and in any case it pulls the dependencies it needs.

→ to remove

Comment (sebalix) [reply 12.0/base_requirements.txt]: Didn't know about ruamel, I said nothing then :)

Comment (tschanzt) [reply 12.0/base_requirements.txt]: dependency dropped for v12

**Comment by guewen** [comment]
Thanks :)

**Pull request merged**